packages/lesswrong/server/callbacks/votingCallbacks.ts

```diff
@@ -47,3 +50,19 @@ voteCallbacks.castVoteAsync.add(async function updateNeedsReview (document: Vote
     void Users.update({_id:voter._id}, {$set:{needsReview: true}})
   }
 });
+
+
+postPublishedCallback.add(async (publishedPost: DbPost) => {
+  // When a post is published (undrafted), update its score. (That is, recompute
+  // the time-decaying score used for sorting, since the time that's computed
+  // relative to has just changed).
+  //
+  // To do this, we mark it `inactive:false` and update the scores on the
+  // whole collection. (This is already something being done frequently by a
+  // cronjob.)
+  if (publishedPost.inactive) {
+    await Posts.update({_id: publishedPost._id}, {$set: {inactive: false}});
+  }
+  
+  await batchUpdateScore({collection: Posts});
```

Comment:
I guess that's one way to do this. I feel like you could have also reused the code that we use for voting: https://github.com/ForumMagnum/ForumMagnum/blob/2260de4d41bb52343727de14cd4fab599d0ce990/packages/lesswrong/server/voteServer.js#L40

Reply:
Oh interesting; I saw that `updateScores.ts` had `updateScore` and `batchUpdateScore`, observed that `updateScore` was dead code, and inferred that they had probably gotten out of sync (or would eventually get out of sync) and that `batchUpdateScore` was the way to go. I somehow overlooked that they're both calling `recalculateScore` under the hood.